### DISCUSSION TITLE
Right align last element on `StackPanel`

### DISCUSSION THREAD

**Discussion by tom-huntington** [microsoft/microsoft-ui-xaml#10136] [Q&A] [answered]
`HorizontalAlignment="Right"` does not work in the following
```
<StackPanel Orientation="Horizontal" >
    <TextBlock VerticalAlignment="Center" Text="Enter Number"/>
    <NumberBox HorizontalAlignment="Right"/>
</StackPanel>
```

What do people do? Use a grid with

```
<Grid.ColumnDefinitions>
    <ColumnDefinition Width="*"/>
    <ColumnDefinition Width="auto"/>
</Grid.ColumnDefinitions>
```

**Comment by Lightczx** [comment]
You can always write a custom Panel to achieve this,
just implement measure & arrange logic by yourself

**Reply by tom-huntington** [reply]
Are you sure? When I looked into the source code, it didn't appear they exposed the ability to override measure.

**Reply by Lightczx** [reply, marked as answer]
StackPanel can't be inherited, you need to inherit from Panel and write all logic yourself.

That's a example write by our team
https://github.com/DGP-Studio/Snap.Hutao/blob/main//src/Snap.Hutao/Snap.Hutao/UI/Xaml/Control/Panel/EqualPanel.cs#L11
